bindings/python/pymongoarrow/monkey.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pymongoarrow.api import (
+    aggregate_arrow_all, find_arrow_all)
+
+
+__all__ = [
+    'patch_all'
+]
+
+
+def patch_all():
+    for method in (aggregate_arrow_all, find_arrow_all):
+        if not hasattr(method, "__target__"):
+            raise ValueError(
+                "cannot patch method with unset '__target__' attribute")
+
+        target_spec = method.__target__
```

Comment:
This seems a bit overly complex. Is there any downside to this approach:
```python
from pymongo.collection import Collection
for method in (aggregate_arrow_all, find_arrow_all):
    setattr(Collection, method.__name__, method)

Reply:
The downside is that we cannot pass the module name/path to be imported as a variable to the vanilla `import` statement. This will become a factor if/when we decide to patch other classes (e.g. `Database.aggregate_arrow_all`). I think this is a more general solution.

Reply:
If we go with your solution we might as well also get rid of the whole business with the decorator and hardcode everything that needs to be patched. I am fine with that solution, but I thought this was more elegant as it allows us to simply apply the decorator to any function that needs to be patched onto one of PyMongo's APIs. I think it will come in really handy as we add more functionality to this project (e.g. `Collection.find_arrow_batches`)

Reply:
I would strongly prefer the _less_ general approach since that's all we need right now. Also if we wanted to patch Database that would also be quite straightforward:
```
from pymongo.database import Database
Database.aggregate_arrow_all = aggregate_arrow_all
```

Reply:
Changed to use what you suggested.

Reply:
Thanks, keeping this simple makes the code easier to read and lessens the risk of bugs.